### ISSUE TITLE
When installing the 7.1.2 the msi asks for the 7.1.0 msi 

### ISSUE THREAD

**Issue by HerraHak** [PowerShell/PowerShell#14919]
When trying to install the 7.1.2, the installer asks me for the path to the 7.1.0 msi installer to uninstall that version.

I had to download it  but then the 7.1.2 installer fails saying it cannot remove the older version.

**Comment by adityapatwardhan** [comment]
@HerraHak Would it be possible for you to provide a repro with either a GIF or a video showing the installation issues.

**Comment by iSazonov** [comment]
I guess the OS was damaged.
We should check the scenario on clean system.

**Comment by HerraHak** [comment]
I successfully updated powershell on another pc so I guess there might be something wrong on the pc I tried first.
As for the video or the gif it's surprising you ask.  I would have thought logs would have been more appropriate.
I'll retry on the original pc where I had the error and try to capture the screen.

**Comment by HerraHak** [comment]
![powershell7 2 install](https://user-images.githubusercontent.com/39991942/110247438-9f418980-7f6c-11eb-8840-93d18062511b.gif)
This is a gif I captured from installing the new version of powershell.

**Comment by SteveL-MSFT** [comment]
I believe if the 7.1.x MSI pkg is not able to uninstall the previous version, it relies on the previous version MSI installer to do the uninstall.  I haven't seen any reports of this issue til now and wonder if something on your system is corrupted.

**Comment by DHowett** [comment]
Yes, it seems that the Installer cache has been corrupted. The MSI installer keeps a copy of every MSI’s metadata stored in a system folder for uninstall purposes. If that is broken, uninstall can fail.

**Comment by HerraHak** [comment]
I wonder what could have been corrupted.  I haven't remove or cleaned any cache of the sort.  This is how I always installed powershell.

Does anyone have any idea on how to fix it?

Thanks

**Comment by iSazonov** [comment]
You can remove the key from Registry manually or use Windows Installer CleanUp Utility. (It is unsupported.)

**Comment by HerraHak** [comment]
I used this utility and the 7.1.2 is now installed on my machine.  Crazy how mystical things can become when Windows Installer start going weird.
https://support.microsoft.com/en-us/topic/fix-problems-that-block-programs-from-being-installed-or-removed-cca7d1b6-65a9-3d98-426b-e9f927e1eb4d
